mqtt-tester/src/client_report.rs

```diff
@@ -95,26 +108,27 @@ async fn check_receiving_server_packet(client_exe_path: &Path) -> miette::Result
         .await
         .map(crate::command::Command::new)?
         .wait_for_write([
-            crate::command::ClientCommand::Send(&[
-                0b0010_0000, // CONNACK
-                0b0000_0010, // Remaining length
-                0b0000_0000, // No session present
-                0b0000_0000, // Connection accepted
-            ]),
-            crate::command::ClientCommand::Send(&[
-                0b1000_0010, // SUBSCRIBE packet
-                0b0000_1000, // Length
-                // Now the variable header
-                0b0000_0000, // Packet ID
-                0b0000_0001,
-                // First sub
-                0b0000_0000,
-                0b0000_0011, // Length
-                b'a',
-                b'/',
-                b'b',
-                0b0000_0001, // QoS
-            ]),
+            crate::command::ClientCommand::Send(
+                crate::util::packet_to_vec(MPacket::Connack({
+                    MConnack {
+                        session_present: false,
+                        connect_return_code: MConnectReturnCode::Accepted,
+                    }
+                }))
+                .await?,
+            ),
+            crate::command::ClientCommand::Send(
+                crate::util::packet_to_vec(MPacket::Subscribe({
+                    MSubscribe {
+                        id: MPacketIdentifier(1),
+                        subscriptions: MSubscriptionRequests {
+                            count: 1,
+                            data: &[b'a', b'/', b'b'],
+                        },
+                    }
```

Comment:
Still not sure about this bit.

Reply:
Looks good to me. You should be able to write `b"a/b"` to get a byte string.